analysis/src/main.rs

```diff
@@ -23,57 +16,52 @@ fn main() -> Result<(), Box<dyn std::error::Error>> {
         .open()
         .unwrap();
 
-    let mut ethernet_frame: ethernet_frame::EthernetFrame = Default::default();
-    let mut ipv4: ip_protocol::IPv4 = Default::default();
-    let _ipv6: ip_protocol::IPv6 = Default::default();
-    let mut counter: u16 = 1;
-    ctrlc::set_handler(|| {
-        KILL_EXECUTION.store(true, std::sync::atomic::Ordering::Relaxed);
-    })?;
+    for _ in 0.. {
+        if let Ok(packet) = cap.next_packet() {
+            let eth_frame = EthernetFrame::serialize(&packet);
 
-    while let Ok(packet) = cap.next_packet() {
-        println!("received packet number: {:?}", counter);
-        ethernet_frame.dest_mac = [
-            packet[0], packet[1], packet[2], packet[3], packet[4], packet[5],
-        ];
-        ethernet_frame.src_mac = [
-            packet[6], packet[7], packet[8], packet[9], packet[10], packet[11],
-        ];
-        ethernet_frame.ip_type = u16::from_be_bytes([packet[12], packet[13]]);
-        ipv4.src_addr = [packet[30], packet[31], packet[32], packet[33]];
-        ipv4.dest_addr = [packet[34], packet[35], packet[36], packet[37]];
+            let ip_packet: Option<Box<dyn Packet>> = match eth_frame.ip_type {
+                EtherType::IPv4 => Some(ip_protocol::IPv4::serialize(&packet, eth_frame)),
+                _ => {
+                    eprintln!("Unknown {:?}", eth_frame.ip_type.to_string());
+                    continue;
+                }
+            };
 
-        conn.execute(
-            "INSERT INTO Packet (DestMac, SrcMac, IpType, SrcAddress, DestAddress) VALUES (?1, ?2, ?3, ?4, ?5)",
-            (
-                ethernet_frame.to_string_dest_mac(),
-                ethernet_frame.to_string_src_mac(),
-                ethernet_frame.to_string_ip_type(),
-                ip_protocol::ToString::src_addr(&ipv4),
-                ip_protocol::ToString::dest_addr(&ipv4)
-            ),
-        )?;
+            // TODO: Insert into database
```

Comment:
Du fjernede hvor det bliver sat i databasen for en comment?

Reply:
Skal gøres i #10 (database trait til `packet`) da IPv4 først er færdigt nu - Der er her `ip_protocol::IPv4::Database::Insert()` skal være